lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -2695,6 +2691,12 @@ FuncInfo* PostVisitFunction(ParseNode* pnode, ByteCodeGenerator* byteCodeGenerat
         return top;
     }
 
+    if (top->paramScope && top->paramScope->GetCanMergeWithBodyScope())
+    {
+        Scope::MergeParamAndBodyScopes(pnode);
+        Scope::RemoveParamScope(pnode);
+    }
```

Comment:
This is necessary for the change, yes. We need to do the merge earlier so we can propagate the has-local-in-closure property to the body scope before we process it.